### PR TITLE
Add site extension generation code

### PR DESCRIPTION
**Pull request by pakrym**
I mostly want to merge zip fallback archive generation so I can test extension creation better and start putting it on CI.

### CODE REVIEW

File: build/SiteExtension.targets [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<Project>
+
+  <Target Name="BuildSiteExtension">
+    <Error Text="Please specify fallback archive zip file localtion via FallbackArchive parameter." Condition="'$(FallbackArchive)' == ''"/>
+    <Error Text="Please specify runtime cache zip file location localtion via RuntimeCache parameter." Condition="'$(RuntimeCache)' == ''"/>
```
Comment (JunTaoLuo) [inline build/SiteExtension.targets]: nit: RuntimeStore?

File: build/repo.targets [hunk 2]
```diff
@@ -285,7 +294,15 @@
     <PropertyGroup>
         <PublishDir Condition="'$(PublishShare)' != ''">$(PublishShare)\fallbackArchives</PublishDir>
     </PropertyGroup>
-    <Copy SourceFiles="$(RepositoryRoot)artifacts\$(TimestampOutputPackageName).lzma;$(RepositoryRoot)artifacts\$(NoTimestampOutputPackageName).lzma" DestinationFolder="$(PublishDir)" Condition="'$(PublishDir)' != ''" />
+
+  <ItemGroup>
+      <FallbackArchiveFile Include="$(RepositoryRoot)artifacts\$(TimestampOutputPackageName).lzma"/>
```
Comment (JunTaoLuo) [inline build/repo.targets]: you can put these all in one include separated by `;`

Comment (pakrym) [inline build/repo.targets]: I was trying to avoid 200 character long line.

Comment (JunTaoLuo) [inline build/repo.targets]: you can put them on separate lines

Comment (JunTaoLuo) [inline build/repo.targets]: https://github.com/aspnet/Coherence-Signed/blob/dev/build/Sign.targets#L17-L22

File: build/repo.targets [hunk 3]
```diff
@@ -285,7 +294,16 @@
     <PropertyGroup>
         <PublishDir Condition="'$(PublishShare)' != ''">$(PublishShare)\fallbackArchives</PublishDir>
     </PropertyGroup>
-    <Copy SourceFiles="$(RepositoryRoot)artifacts\$(TimestampOutputPackageName).lzma;$(RepositoryRoot)artifacts\$(NoTimestampOutputPackageName).lzma" DestinationFolder="$(PublishDir)" Condition="'$(PublishDir)' != ''" />
+
+  <ItemGroup>
```
Comment (JunTaoLuo) [inline build/repo.targets]: nit: in fact you don't need this itemgroup and should be able to just specify the files in the Copy target.

**Comment by JunTaoLuo** [comment]
:shipit: if it works. Not sure why the builds are failing though?

**Comment by pakrym** [comment]
:up::date:

**Comment by Tratcher** [comment]
You've taken a prime package name (Microsoft.AspNetCore.SiteExtension) for test infrastructure... And the name doesn't tell you what it's for. AzureTestDepoyer?

**Pull request merged**